### ISSUE TITLE
Decouple e2e test framework from mocha

### ISSUE THREAD

**Issue by thomaseizinger** [comit-network/comit-rs#833]
Currently, the e2e testing framework uses `mocha` primitives, such as `it` and `describe` to build up a test suite that executes the individual steps. This solution is sub-optimal for the following reasons:

- mocha uses global state which makes it hard to trace, what the `describe` and `it` invocations will actually do
- mocha needs a workaround for long-running setup tasks (as we need them for the erc20 test to deploy the contract) which requires calling `run` once the setup is done for mocha know that it can start executing the tests.
- each step in a swap is its individual test (from a mocha perspective) whereas conceptually, doing one swap is a test

We should revise the implementation so that the actual execution of a swap is not depending on mocha for executing (but for example just goes through a list of closures and runs them). Given that, it should be easy to use mocha to define tests like "btc-eth happy path swap" that uses above mentioned library to execute the swap.

**Comment by LLFourn** [comment]
>...easy to use mocha to define tests like `btc-eth happy path swap` that uses above mentioned library to execute the swap.

Can you explain what you mean by `btc-eth happy path swap` is that an example of a cli command or ...?

This is a good idea anyways.

**Comment by thomaseizinger** [comment]
> Can you explain what you mean by btc-eth happy path swap is that an example of a cli command or ...?

No it is not a CLI command. Probably should not have used backticks here. This was meant as an example for a test:

```js
describe("btc-eth", () => {
    it("happy path swap", () => {
		// do the swap here with the library
	})
})
```

Once we have that, it should be easy to selectively run tests from the commandline without any bash-fu.

**Comment by D4nte** [comment]
I think we should close this in favor of #985 (maybe put a comment in #985 about it)

**Comment by bonomat** [comment]
closed in favor of #985
